File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution1 {   //Brute force
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int>ans;
        for(int i=0;i<nums.size();i++){
            int pr=1;
            for(int j=0;j<nums.size();j++){
                if(i!=j){
                    pr*=nums[j];
                }
            }
            ans.push_back(pr);
        }
        return ans;
    }
};
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums){
       long long pr=1;
       int zero=0;
       int n=nums.size();
       for(int i=0;i<nums.size();i++){
        if(nums[i]==0) zero++;
        else pr*=nums[i];
       }
       vector<int>ans(n,0);
       if(zero>1) return ans;
       else if(zero==1){
        for(int i=0;i<n;i++){
            if(nums[i]==0){
                ans[i]=pr;
                break;
            }
        }
       }
        else{
            for(int i=0;i<n;i++){
                if(nums[i]!=0){
                    ans[i]=pr/nums[i];
                }
            }
        }
        return ans;

    }

};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
#bruete Force
class Solution1:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        list1 = []
        for i in range(len(nums)):
            pr=1
            for j in range(len(nums)):
                if j != i:
                    pr *= nums[j]
            
            list1.append(pr)
        
        return list1

class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n=len(nums)
        pr = 1
        count =0
        for i in range(n):
            if nums[i] == 0:
                count +=1
            else:
                pr *=nums[i]
        
        list1 = [0]*n

        if count > 1:
            return list1
        elif count ==1:
            for  i  in range(n):
                if nums[i] == 0:
                    list1[i] = pr
        
        else:
            for i in range(n):
                list1[i] = pr // nums[i]
        
        return list1

        
            
